.travis.yml

```diff
@@ -8,6 +8,12 @@ matrix:
       addons: {apt: {packages: [cabal-install-1.22, ghc-7.10.3], sources: [hvr-ghc]}}
     - env: CABALVER=1.24 GHCVER=8.0.1
       addons: {apt: {packages: [cabal-install-1.24, ghc-8.0.1],  sources: [hvr-ghc]}}
+    - env: CABALVER=1.18 GHCVER=7.8.4 DYNAMIC=--enable-executable-dynamic
+      addons: {apt: {packages: [cabal-install-1.18, ghc-7.8.4],  sources: [hvr-ghc]}}
+    - env: CABALVER=1.22 GHCVER=7.10.3 DYNAMIC=--enable-executable-dynamic
+      addons: {apt: {packages: [cabal-install-1.22, ghc-7.10.3], sources: [hvr-ghc]}}
+    - env: CABALVER=1.24 GHCVER=8.0.1 DYNAMIC=--enable-executable-dynamic
+      addons: {apt: {packages: [cabal-install-1.24, ghc-8.0.1],  sources: [hvr-ghc]}}
```

Comment:
continuing the discussion... it's probably enough to test just one of these configurations